WASM: `EXPORTED_FUNCTIONS` does not work as intended

if one tries to export C API functions in the wasm binary, configure fails:

**without**
```
cargo@workbook:~/cvc5$ ./configure.sh --static --static-binary --auto-download --wasm=JS --wasm-flags='-s EXPORTED_FUNCTIONS=_main' --name=prod 
configure: cmake /home/cargo/cvc5 -DPython_EXECUTABLE=/usr/bin/python3 -DENABLE_AUTO_DOWNLOAD=ON -DBUILD_SHARED_LIBS=OFF -DSTATIC_BINARY=ON -DUSE_POLY=ON -DCMAKE_PREFIX_PATH= -DPYTHONIC_PATH= -DWASM=JS "-DWASM_FLAGS=-s EXPORTED_FUNCTIONS=_main" -DCMAKE_TOOLCHAIN_FILE=/usr/share/emscripten/cmake/Modules/Platform/Emscripten.cmake -DCMAKE_CROSSCOMPILING_EMULATOR=/usr/bin/node
-- Found Git: /usr/bin/git (found version "2.47.3")
-- Building Production build
-- Performing Test HAVE_C_FLAG_O3
-- Performing Test HAVE_C_FLAG_O3 - Success
-- Configuring with C flag '-O3'
-- Performing Test HAVE_CXX_FLAG_O3
-- Performing Test HAVE_CXX_FLAG_O3 - Success
-- Configuring with CXX flag '-O3'
-- Performing Test HAVE_C_FLAG_Wall
```

**with**
```
cargo@workbook:~/cvc5$ ./configure.sh --static --static-binary --auto-download --wasm=JS --wasm-flags='-s EXPORTED_FUNCTIONS=_cvc5_cmd_invoke' --name=prod 
configure: cmake /home/cargo/cvc5 -DPython_EXECUTABLE=/usr/bin/python3 -DENABLE_AUTO_DOWNLOAD=ON -DBUILD_SHARED_LIBS=OFF -DSTATIC_BINARY=ON -DUSE_POLY=ON -DCMAKE_PREFIX_PATH= -DPYTHONIC_PATH= -DWASM=JS "-DWASM_FLAGS=-s EXPORTED_FUNCTIONS=_cvc5_cmd_invoke" -DCMAKE_TOOLCHAIN_FILE=/usr/share/emscripten/cmake/Modules/Platform/Emscripten.cmake -DCMAKE_CROSSCOMPILING_EMULATOR=/usr/bin/node
-- Found Git: /usr/bin/git (found version "2.47.3")
-- Building Production build
-- Performing Test HAVE_C_FLAG_O3
-- Performing Test HAVE_C_FLAG_O3 - Failed
-- Performing Test HAVE_CXX_FLAG_O3
-- Performing Test HAVE_CXX_FLAG_O3 - Failed
-- Performing Test HAVE_C_FLAG_Wall
```

The reason is that the **Performing Test** steps call the linker with all options and the linker fails to export C API functions (here: cvc5_cmd_invoke) which it cannot find in the test binary

```
  -
    kind: "try_compile-v1"
    backtrace:
      - "/usr/share/cmake-3.31/Modules/Internal/CheckSourceCompiles.cmake:108 (try_compile)"
      - "/usr/share/cmake-3.31/Modules/Internal/CheckCompilerFlag.cmake:18 (cmake_check_source_compiles)"
      - "/usr/share/cmake-3.31/Modules/CheckCCompilerFlag.cmake:56 (cmake_check_compiler_flag)"
      - "cmake/Helpers.cmake:76 (check_c_compiler_flag)"
      - "cmake/Helpers.cmake:93 (add_check_c_flag)"
      - "CMakeLists.txt:239 (add_check_c_cxx_flag)"
    checks:
      - "Performing Test HAVE_C_FLAG_O3"
    directories:
      source: "/home/cargo/cvc5/CMakeFiles/CMakeScratch/TryCompile-cMSZrF"
      binary: "/home/cargo/cvc5/CMakeFiles/CMakeScratch/TryCompile-cMSZrF"
    cmakeVariables:
      CMAKE_CXX_COMPILER_CLANG_SCAN_DEPS: "/usr/bin/clang-scan-deps-19"
      CMAKE_C_FLAGS: ""
      CMAKE_C_FLAGS_DEBUG: "-g"
      CMAKE_EXE_LINKER_FLAGS: " -s MODULARIZE -s EXPORT_NAME=CvcModule -s EXPORTED_FUNCTIONS=_main,_cvc5_cmd_invoke"
      CMAKE_MODULE_PATH: "/home/cargo/cvc5/cmake"
    buildResult:
      variable: "HAVE_C_FLAG_O3"
      cached: true
      stdout: |
        Change Dir: '/home/cargo/cvc5/CMakeFiles/CMakeScratch/TryCompile-cMSZrF'
        
        Run Build Command(s): /usr/bin/cmake -E env VERBOSE=1 /usr/bin/gmake -f Makefile cmTC_22137/fast
        /usr/bin/gmake  -f CMakeFiles/cmTC_22137.dir/build.make CMakeFiles/cmTC_22137.dir/build
        gmake[1]: Entering directory '/home/cargo/cvc5/CMakeFiles/CMakeScratch/TryCompile-cMSZrF'
        Building C object CMakeFiles/cmTC_22137.dir/src.c.o
        /usr/share/emscripten/emcc -DHAVE_C_FLAG_O3  -std=gnu11   -O3 -MD -MT CMakeFiles/cmTC_22137.dir/src.c.o -MF CMakeFiles/cmTC_22137.dir/src.c.o.d -o CMakeFiles/cmTC_22137.dir/src.c.o -c /home/cargo/cvc5/CMakeFiles/CMakeScratch/TryCompile-cMSZrF/src.c
        Linking C executable cmTC_22137.js
        /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_22137.dir/link.txt --verbose=1
        wasm-ld-19: error: symbol exported via --export not found: cvc5_cmd_invoke
        emcc: error: '/usr/bin/wasm-ld-19 -o cmTC_22137.wasm CMakeFiles/cmTC_22137.dir/src.c.o -L/usr/share/emscripten/cache/sysroot/lib/wasm32-emscripten -lGL-getprocaddr -lal -lhtml5 -lstubs-debug -lnoexit -lc-debug -ldlmalloc -lcompiler_rt -lc++-noexcept -lc++abi-debug-noexcept -lsockets -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr /tmp/tmp5hj4nry3libemscripten_js_symbols.so --strip-debug --export=cvc5_cmd_invoke --export=emscripten_stack_get_end --export=emscripten_stack_get_free --export=emscripten_stack_get_base --export=emscripten_stack_get_current --export=emscripten_stack_init --export=_emscripten_stack_alloc --export=__get_temp_ret --export=__set_temp_ret --export=_emscripten_tempret_set --export=_emscripten_tempret_get --export=__wasm_call_ctors --export=_emscripten_stack_restore --export-if-defined=__start_em_asm --export-if-defined=__stop_em_asm --export-if-defined=__start_em_lib_deps --export-if-defined=__stop_em_lib_deps --export-if-defined=__start_em_js --export-if-defined=__stop_em_js --export-if-defined=main --export-if-defined=__main_argc_argv --export-if-defined=fflush --export-table -z stack-size=65536 --no-growable-memory --initial-heap=16777216 --no-entry --stack-first --table-base=1' failed (returned 1)
        /usr/share/emscripten/emcc  -s MODULARIZE -s EXPORT_NAME=CvcModule -s EXPORTED_FUNCTIONS=_main,_cvc5_cmd_invoke  @CMakeFiles/cmTC_22137.dir/objects1.rsp -o cmTC_22137.js
        gmake[1]: *** [CMakeFiles/cmTC_22137.dir/build.make:104: cmTC_22137.js] Error 1
        gmake[1]: Leaving directory '/home/cargo/cvc5/CMakeFiles/CMakeScratch/TryCompile-cMSZrF'
        gmake: *** [Makefile:134: cmTC_22137/fast] Error 2
        
      exitCode: 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -645,6 +645,11 @@ else()
   endif()
 endif()
 
+# Re-apply EXPORTED_FUNCTIONS after configure checks complete
+if(NOT "${WASM_EXPORTED_FUNCTIONS_FLAG}" STREQUAL "")
+  string(APPEND CMAKE_EXE_LINKER_FLAGS " ${WASM_EXPORTED_FUNCTIONS_FLAG}")
+endif()
+
 #-----------------------------------------------------------------------------#
 # Add subdirectories
 

